pocs/tests/test_social_messaging.py

```diff
@@ -0,0 +1,103 @@
+import pytest
+import tweepy
+import requests
+import unittest.mock
+
+from pocs.utils.social_twitter import SocialTwitter
+from pocs.utils.social_slack import SocialSlack
+
+
+@pytest.fixture(scope='module')
+def twitter_config():
+    twitter_config = {'consumer_key': 'mock_consumer_key', 'consumer_secret': 'mock_consumer_secret', 'access_token': 'mock_access_token', 'access_token_secret': 'access_token_secret'}
+    return twitter_config
+
+
+@pytest.fixture(scope='module')
+def slack_config():
+    slack_config = {'webhook_url': 'mock_webhook_url', 'output_timestamp': False}
+    return slack_config
+
+
+# Twitter sink tests
+def test_no_consumer_key(twitter_config):
+    with unittest.mock.patch.dict(twitter_config):
+        del twitter_config['consumer_key']
+        with pytest.raises(ValueError) as ve:
+            social_twitter = SocialTwitter(**twitter_config)
+        assert 'consumer_key parameter is not defined.' == str(ve.value)
+
+
+def test_no_consumer_secret(twitter_config):
+    with unittest.mock.patch.dict(twitter_config):
+        del twitter_config['consumer_secret']
+        with pytest.raises(ValueError) as ve:
+            social_twitter = SocialTwitter(**twitter_config)
+        assert 'consumer_secret parameter is not defined.' == str(ve.value)
+
+
+def test_no_access_token(twitter_config):
+    with unittest.mock.patch.dict(twitter_config):
+        del twitter_config['access_token']
+        with pytest.raises(ValueError) as ve:
+            social_twitter = SocialTwitter(**twitter_config)
+        assert 'access_token parameter is not defined.' == str(ve.value)
+
+
+def test_no_access_token_secret(twitter_config):
+    with unittest.mock.patch.dict(twitter_config):
+        del twitter_config['access_token_secret']
+        with pytest.raises(ValueError) as ve:
+            social_twitter = SocialTwitter(**twitter_config)
+        assert 'access_token_secret parameter is not defined.' == str(ve.value)
+
+
+def test_send_message_twitter(twitter_config):
+    with unittest.mock.patch.object(tweepy.API, 'update_status') as mock_update_status:
+        social_twitter = SocialTwitter(**twitter_config)
+        mock_message = "mock_message"
+        mock_timestamp = "mock_timestamp"
+        social_twitter.send_message(mock_message, mock_timestamp)
+
+        mock_update_status.assert_called_once_with('{} - {}'.format(mock_timestamp, mock_message))
+
+
+def test_send_message_twitter_no_timestamp(twitter_config):
+    with unittest.mock.patch.dict(twitter_config, {'output_timestamp': False}):
```

Comment:
FYI, for this kind of nested `with` you can put multiple `with_items` in a single with statement. See the docs here: https://docs.python.org/3/reference/compound_stmts.html#the-with-statement

Reply:
Will change this thanks